### DISCUSSION TITLE
Can we add an external geom in the top right of ggside?

### DISCUSSION THREAD

**Discussion by xiangpin** [jtlandis/ggside#53] [Ideas]
Can we add an external geom such as geom_text in the top right of `ggside`?

![捕获](https://github.com/jtlandis/ggside/assets/17870644/13313955-a47f-48ed-b2d3-f1b1d6d008f6)

**Comment by jtlandis** [comment]
This is something that I have put some thought into but I personally do not know what conventions I should enforce in this graph or how to formalize it.

As a rule of thumb, the `geom_xside*()` functions will share the x axis of the main panel plot, and the `geom_yside*()` functions will share the y-axis of the main panel plot. This means that `geom_xside*()`'s y-axis is allowed to be free from the y axis of the main plot and vise versa for the `geom_yside*()`'s x-axis.  The question then becomes, how should this dependency be applied to the corner plotting range?

If both geom_yside and geom_xside exit on the plot, then should this new panel follow the conventions in place to the main panel? this would have an effect similar to `facet_grid` where each row/column of the panel layouts share the same scale.

I'm open to starting a discussion on this topic, but these are the questions I would like to have an idea about before I put time into creating this feature.

As a side note for my own curiosity, what type of data would we plot in this corner? I personally couldn't think of meaningful summary statistics for that panel, but I'm curious if you have an example!

P.S. Sorry this issue has become a bit stale, I want to make some updates within the next two months to this project.

**Comment by jtlandis** [comment]
I will be converting this issue into a discussion as I genuinely want feedback from the community of users and how they might utilize this feature.
